src/web/source.rs

```diff
@@ -217,8 +243,8 @@ pub fn source_browser_handler(req: &mut Request) -> IronResult<Response> {
         (None, false)
     };
 
-    let file_list =
-        FileList::from_path(&mut conn, &name, &version, &req_path).ok_or(Nope::NoResults)?;
+    let file_list = FileList::from_path(&mut conn, &crate_name, &version, &req_path)
+        .ok_or(Nope::ResourceNotFound)?;
```

Comment:
This is what fixed the 500 error; before it was treating this as a search page and panicking because variables weren't set.